### PR TITLE
Fix bgrewriteaof test

### PR DESCRIPTION
**Pull request by mikemagowan**
Fix to the t/backend.t test for bgrewriteaof.  Checks to make sure that the lastsave value has changed after calling bgsave before running the bgrewriteaof call.  Have attached a sensible timeout; it waits no longer than 5 seconds.

Also checks for both possible responses from bgrewriteaof to be on the safe side.

### CODE REVIEW

File: t/backend.t [hunk 1]
```diff
@@ -51,11 +51,22 @@ for my $method (qw( flushall flushdb resetstat save )) {
 }
 
 {
+  my $lastsave = $backend->lastsave;
+  my $loopcount;
+
   @res = $backend->bgsave;
   like $res[0], qr{start}i, 'bgsave started';
 
+  while ($backend->lastsave != $lastsave) {
```
Comment (Grinnz) [inline t/backend.t]: Shouldn't this be "while ($backend->lastsave == $lastsave)"?

File: t/backend.t [hunk 2]
```diff
@@ -51,11 +51,22 @@ for my $method (qw( flushall flushdb resetstat save )) {
...
+    sleep 1;
```
Comment (Grinnz) [inline t/backend.t]: Can probably use Time::HiRes to sleep in smaller intervals here to be nice to testers (it usually is not going to take very long).

File: t/backend.t [hunk 3]
```diff
@@ -51,11 +51,22 @@ for my $method (qw( flushall flushdb resetstat save )) {
...
+    $loopcount++;
+    last if $loopcount > 4; # Wait max 5 secs
+  }
+
   @res = $backend->bgrewriteaof;
-  like $res[0], qr{scheduled}i, 'bgrewriteaof scheduled';
+  like $res[0], qr{scheduled|started}i, 'bgrewriteaof scheduled/started';
+
+  sleep 1; # Give the rewrite chance to complete
```
Comment (Grinnz) [inline t/backend.t]: Might be able to check fields in $redis->info('persistence') to find out when this completes, like earlier in the test file?

**Comment by mikemagowan** [comment]
Ok - how about this.  It's cleaner and more consistent now I think.

**Comment by jhthorsen** [comment]
I think @Grinnz is better suited to merge this than me, but I welcome the improvement! 👍

File: t/backend.t [hunk 4]
```diff
@@ -47,15 +48,24 @@ SKIP: {
 for my $method (qw( flushall flushdb resetstat save )) {
   @res = $backend->$method;
   is $res[0], 'OK', $method;
-  sleep 1 while $backend->info('persistence')->{aof_rewrite_in_progress};
+  sleep 0.1 while $backend->info('persistence')->{aof_rewrite_in_progress};
 }
 
 {
+  my $lastsave = $backend->lastsave;
```
Comment (Grinnz) [inline t/backend.t]: These lines probably aren't needed now. Otherwise LGTM

**Pull request merged**